server/server_test.go

```diff
@@ -382,7 +382,7 @@ func TestDirectOK(t *testing.T) {
 		}
 
 		resp, _ := http.ReadResponse(bufio.NewReader(conn), nil)
-		assert.Equal(t, http.StatusBadGateway, resp.StatusCode, "should fail")
+		assert.Equal(t, http.StatusBadRequest, resp.StatusCode, "should fail")
```

Comment:
Okay, I figured out the failing tests.  It was actually due to a change in the behavior of the http package introduced in Go 1.12.  I'm not sure if this is the appropriate fix - do we want to return Bad Gateway specifically or do we just want to return any sensible error status?

This is orthogonal to the Go mod changes and is instead just related to Go 1.12 support.  If we need to do something more complicated than this drive-by fix, maybe I should separate this into its own PR.

Reply:
Here's the relevant part of the Go 1.12 changelog: https://golang.org/doc/go1.12#net/http

Reply:
Good catch! It seems to me that this change is semantically more in line with what we were intending to check - 	
```// HTTPS can't be tunneled using Direct Proxying, as redirections
// require a TLS handshake between the proxy and the origin```
It seems a 400 BadRequest error code would be more appropriate to check for than 502 BadGateway in this case.

Reply:
Agreed!